### PR TITLE
[IMP] im_livechat: unpin livechat channel after one day

### PR DESCRIPTION
**Pull request by pka-odoo**
PURPOSE:

livechat channel doesn't unpinned after one day automatically,
even if operator have seen that channel or not. it can be
case where operator is out of office for long and didn't check
the messages

SPECIFICATION:

livechat channel should be unpinned after one day of operator seen
that channel.

task-2446972

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/im_livechat/models/mail_channel_partner.py [hunk 1]
```diff
@@ -18,6 +20,6 @@ def _gc_unpin_livechat_sessions(self):
                 SELECT cp.id FROM mail_channel_partner cp
                 INNER JOIN mail_channel c on c.id = cp.channel_id
                 WHERE c.channel_type = 'livechat' AND cp.is_pinned is true AND
-                    cp.write_date < current_timestamp - interval '1 day'
+                    cp.last_seen_date < current_timestamp - interval '1 day'
```
Comment (seb-odoo) [inline addons/im_livechat/models/mail_channel_partner.py:23]: Also please take into account that the spec has been updated to only unpin after a day but also only if there are no unread messages for this channel partner.

**Review by seb-odoo — CHANGES_REQUESTED** [review]
Thank you for the PR. Quick check below

File: addons/im_livechat/models/mail_channel_partner.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models
+from odoo import api, fields, models
 
 
 class ChannelPartner(models.Model):
     _inherit = 'mail.channel.partner'
 
+    last_seen_date = fields.Datetime("Last seen date")
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/im_livechat/models/mail_channel_partner.py:10]: Would probably call it `last_seen_dt` instead (it's a datetime, not just a date),  similar naming than `last_interest_dt`

**Comment by C3POdoo** [comment]
Dear @pka-odoo,

Thank you for your contribution but we are closing it due to inactivity.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**